internal/mocks/connection.go

```diff
@@ -0,0 +1,298 @@
+package mocks
+
+import (
+	"errors"
+	"math"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/Azure/go-amqp/internal/buffer"
+	"github.com/Azure/go-amqp/internal/encoding"
+	"github.com/Azure/go-amqp/internal/frames"
+)
+
+// NewConnection creates a new instance of MockConnection.
+func NewConnection() *MockConnection {
+	return &MockConnection{
+		readData:    make(chan []byte),
+		pendingData: [][]byte{},
+	}
+}
+
+// MockConnection is a mock connection that satisfies the net.Conn interface.
+type MockConnection struct {
+	// FrameCallback is invoked when an AMQP frame is received.
+	// Return a non-nil error to simulate a write error.
+	FrameCallback func(frames.FrameBody) error
+	readDLDur     time.Duration
+	readDL        *time.Timer
+	readData      chan []byte
+	pendingDataMu sync.Mutex
+	pendingData   [][]byte
+	//writeDL *time.Timer
```

Comment:
NIT: remove comment.

But I am curious what you did with it....how cool is this MockConnection going to be? ;)